javav2/example_code/transcribe/src/main/java/com/amazonaws/transcribestreaming/TranscribeMedicalStreamingDemoApp.java

```diff
@@ -0,0 +1,206 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazonaws.transcribestreaming;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.transcribestreaming.TranscribeStreamingAsyncClient;
+import software.amazon.awssdk.services.transcribestreaming.model.*;
+import javax.sound.sampled.*;
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicLong;
+
+// snippet-start:[transcribe.java-medical-streaming-demo]
+/*
+To run this AWS code example, ensure that you have set up your development
+ environment, including your AWS credentials.
+
+ For information, see this documentation topic:
+
+ https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
+
+This code demonstrates the process of starting a medical transcription job using the AWS Transcribe
+Streaming service, including setting up the audio input stream, configuring the transcription request,
+and handling the transcription response.
+ */
+
+public class TranscribeMedicalStreamingDemoApp {
+    private static TranscribeStreamingAsyncClient client;
+
+    public static void main(String args[])
+        throws ExecutionException, InterruptedException, LineUnavailableException {
+
+        client = TranscribeStreamingAsyncClient.builder()
+            .credentialsProvider(getCredentials())
+            .build();
+
+        CompletableFuture<Void> result = client.startMedicalStreamTranscription(getMedicalRequest(16_000),
+            new AudioStreamPublisher(getStreamFromMic()),
+            getMedicalResponseHandler());
+
+        result.get();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like you are using `CompletableFuture.get()` or `Future.get()` without setting its timeout parameter. Be sure to set timeout values or handle timeout exceptions to prevent threads from waiting for a process to end. This can lead to downtime for your customers. For more information, see [CompletableFuture](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#get-long-java.util.concurrent.TimeUnit-) and [Future](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Future.html#get-long-java.util.concurrent.TimeUnit-) documentation.

Suggested remediation:
Set timeout parameters for `CompletableFuture.get()` or `Future.get()`.Exceptions, such as `ExecutionException`, `InterruptedException` and `TimeoutException` are caught and errors are logged.
```
@@ -52,3 +52,14 @@
 
-        result.get();
+        try {
+            result.get(100, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException ie) {
+            log.error(&#34;Thread interrupted.&#34;, ie);
+            throw new InterruptedException(ie.getMessage());
+        } catch (ExecutionException xe) {
+            log.error(&#34;The task aborted when attempting to retrieve its result.&#34;, xe);
+            throw new ExecutionException(xe.getMessage());
+        } catch (TimeoutException te) {
+            log.error(&#34;Future is not done when timeout.&#34;, te);
+            throw new TimeoutException(te.getMessage());
+        }
         client.close();

```